Network/Public/Include/Network/CombiningHandlers.hpp

```diff
@@ -14,36 +14,36 @@ namespace Network
 		public:
 		/**
 		* @brief Method for serialization, compression and encryption of outcoming messages.
-		* @param message    - buffer that contains data that should be serialized, compressed and encrypted.
-		* @param bodyBuffer - buffer that will contain serialized, compressed and encrypted body.
+		* @param source      - buffer that contains data that should be serialized, compressed and encrypted.
+		* @param destination - buffer that will contain serialized, compressed and encrypted body.
 		*/
-		MessageProcessingState operator()(const Message& message,yas::shared_buffer& bodyBuffer) const
+		static MessageProcessingState handleOutcomingMessage(const Message& source,yas::shared_buffer& destination)
 		{
-			auto serializationHandler = std::make_unique<SerializationHandler>();
-			auto compressionHandler   = std::make_unique<CompressionHandler>();
-			auto encryptionHandler    = std::make_unique<EncryptionHandler>();
+			static auto serializationHandler = std::make_unique<SerializationHandler>();
+			static auto compressionHandler   = std::make_unique<CompressionHandler>();
+			static auto encryptionHandler    = std::make_unique<EncryptionHandler>();
 
 			serializationHandler->setNext(compressionHandler.get());
 			compressionHandler-> setNext(encryptionHandler.get());
 
-			return serializationHandler->handleOutcomingMessage(message, bodyBuffer);
+			return serializationHandler->handleOutcomingMessage(source, destination);
 		}
 
 		/**
 		* @brief Method for decryption, decompression and deserialization of incoming message bodies.
-		* @param buffer - buffer that contains data that should be decrypted, decompressed and deserialized.
-		* @param message - variable that will contain decrypted, decompressed and deserialized message body.
+		* @param destination - buffer that contains data that should be decrypted, decompressed and deserialized.
+		* @param source      - variable that will contain decrypted, decompressed and deserialized message body.
 		*/
-		MessageProcessingState operator()(const yas::shared_buffer buffer, Message& message) const
+		static MessageProcessingState handleIncomingMessage(const yas::shared_buffer destination, Message& source)
```

Comment:
Is buffer are destination here?

Reply:
You have yas buffer twice as destination point. Am I wrong, but at deserialisation process we handle buffer to message

Reply:
@Orange-hanter , Yes, you are absolutely right, the destination name for const yas::shared_buffer in the handlncomingMessage method does not make sense, so I suggested changing this name back to a buffer